### ISSUE TITLE
Range slider moves

### ISSUE THREAD

**Issue by nikolas-burkoff** [insightsengineering/teal.slice#184]
If you move the left edge of the slider away from 0 and then back to 0 it will jump out to 0.01 which is confusing.
Please test with this app but also other apps and values of the range to make sure this hasn't caused unforeseen issues.

Probably best to start this one from the refactor branch even though the issue is on main as well

```
library(shiny)
library(teal.slice)
library(shinyjs)

x <- c(0.0091, 0.01 + runif(50, max= 0.99))

my_filter_state <- teal.slice:::init_filter_state(
  x = x,
  varname = "VARNAME",
  varlabel = "A label"
)

#####

ui <- fluidPage(
  useShinyjs(),
  column(3,
         div(  # only need a div so that we can hide the ui when the close button is pressed
           id = "x",
           my_filter_state$ui("fs"),
           h4("Condition (i.e. call)"), # show the condition generated by this FilterState
           textOutput("condition"),
           h4("Formatted state"), # show the human readable state
           textOutput("formatted"),
           h4("Unformatted state"), # show the raw state
           textOutput("unformatted"),
         )
  )
)

server <- function(input, output, session) {

  output$condition <- renderPrint(my_filter_state$get_call())
  output$formatted <- renderText(my_filter_state$format())
  output$unformatted <- renderPrint(my_filter_state$get_state()) # also get_selected(), get_keep_na

  signal_to_remove <- my_filter_state$server("fs")  # filterstate$server returns a trigger when the close button is pressed
  observeEvent(signal_to_remove(), shinyjs::hide("x")) # and shinyjs removes the ui
}

shinyApp(ui, server)
```

**Comment by nikolas-burkoff** [comment]
So I don't know why we actually need to do this:
![image](https://user-images.githubusercontent.com/15201933/216101302-a6a86518-9218-441b-8615-392e1954eafc.png)

Surely `dplyr::filter(df, x > 0.0991)` and `dplyr::filter(df, x > 0)` are the same if there is no data between 0 and 0.0991 and users used the slider to select x = 0.

If we call `self$set_selected(input$selection)` instead of what's done (and remove the warnings and errors from `remove_out_of_bound_values` and `check_in_range` which reset the choices /crash the app respectively then it all seems to be fine

**Comment by chlebowa** [comment]
Based on the app above, I would guess this was to simplify the printing of the filter summary: one sets the slider to 0, but the included range is from 0.0091 (provided that's the lowest value). Get the range once, use it twice, in subset expression and in filter summary. If that's the reason the slider snaps to rounded limit, I'm all for separating the two ranges.

Sorry if this is nonsense, I'm shooting in the dark.

**Comment by chlebowa** [comment]
> User selects range [0, 1] on slider -> triggers observeevent which set selected state to [0.009, 1] which triggers observe event to update slider input given state -> slider is updated to [0.1, 1].

That's roughly what I meant. I'll try and have a look.
